### ISSUE TITLE
Implement process pipeline

### ISSUE THREAD

**Issue by skotz** [briankoser/thomas#12]
To get user input without blocking or continuing execution without the result, I suggest using a promise pipeline. Either a library or roll-your-own.

**Comment by briankoser** [comment]
@skotz I've barely used promises so I'll read up on them tomorrow.

**Comment by briankoser** [comment]
@skotz I'm guessing you want to write this?

**Comment by skotz** [comment]
@briankoser I guess it depends on how you want to manage user interaction with the library. My thoughts are that we either need to rework the methods into something that doesn't require requesting user interaction in the middle of a function that needs the result of that prompt, or we need to wrap the current implementation in a mess of callbacks (or "promises").

**Comment by skotz** [comment]
@briankoser We could model the library to fit the interface instead of writing an interface on the library. Perhaps something like getGameComparison for getting an object containing two games to compare, and an updateGameComparison for reporting which game the user chose. This way we could bind the former to a button or run it on page load, and bind the second to whatever the user clicks to make a choice. The model could be rebuilt after each call to the update. This way we don't need to request the decisions in the middle of building the model.

**Comment by briankoser** [comment]
@skotz :+1: for restructuring the library, that makes more sense. I hadn't thought the UI through yet. I don't have a preference on library vs. roll-your-own, so that's completely up to you.

**Comment by skotz** [comment]
@briankoser Let me know if you have specific naming conventions for code or branch names, and whether or not it's important for me to squash commits so history compiles.

**Comment by briankoser** [comment]
Nope, no real conventions. I'm not worried about squashing commits right now, but that'll probably change when I get around to writing tests.
